packages/resolver/src/adapters/unstoppable/api.ts

```diff
@@ -42,13 +40,33 @@ export const unstoppableApiGetCryptoAddress = (
 
         handleApiError(error)
       } else {
-        const parsedResponse = UnstoppableApiResponseSchema.parse(
+        const safeParsedAdaResponse = UnstoppableApiAdaResponseSchema.safeParse(
           response.value.data,
         )
-
-        const result = parsedResponse.records['crypto.ADA.address']
-        if (!result) throw new Resolver.Errors.NotFound()
-        return result
+        const safeParsedGeneralResponse =
+          UnstoppableApiGeneralResponseSchema.safeParse(response.value.data)
+
+        const hasCardanoAddress = safeParsedAdaResponse.success
+        const hasOtherBlockchainAddress =
+          !hasCardanoAddress &&
+          safeParsedGeneralResponse.success &&
+          Object.keys(response.value.data.records).length > 0
+        const hasNotAnyAddress =
+          !hasOtherBlockchainAddress && safeParsedGeneralResponse.success
+
+        if (hasCardanoAddress) {
+          const result = response.value.data.records['crypto.ADA.address']
+          return result
```

Comment:
it was used before just to throw, now it is throwing early
```suggestion
          return response.value.data.records['crypto.ADA.address']
```